node/core/backing/src/lib.rs

```diff
@@ -74,9 +74,15 @@ enum Error {
 	#[error("Signature is invalid")]
 	InvalidSignature,
 	#[error("Failed to send candidates {0:?}")]
-	Send(Vec<NewBackedCandidate>),
-	#[error("Oneshot never resolved")]
-	Oneshot(#[from] #[source] oneshot::Canceled),
+	Send(Vec<BackedCandidate>),
+	#[error("FetchPoV channel closed before receipt")]
+	FetchPoV(#[source] oneshot::Canceled),
+	#[error("ValidateFromChainState channel closed before receipt")]
+	ValidateFromChainState(#[source] oneshot::Canceled),
+	#[error("StoreAvailableData channel closed before receipt")]
+	StoreAvailableData(#[source] oneshot::Canceled),
+	#[error("a channel was closed before receipt in try_join!")]
+	JoinMultiple(#[source] oneshot::Canceled),
 	#[error("Obtaining erasure chunks failed")]
 	ObtainErasureChunks(#[from] #[source] erasure_coding::Error),
```

Comment:
`#[from]` always includes the `#[source]` attribute. 

> The #[from] attribute always implies that the same field is #[source], so you don't ever need to specify both attributes.

https://docs.rs/thiserror/1.0.21/thiserror

Reply:
b99721d